src/GT.Core/Services/Impl/GTListingService.cs

```diff
@@ -43,106 +45,107 @@ public class GTListingService : IGTListingService
-			newListing.JobTitle = listingDTO.JobTitle;
-			newListing.FTE = listingDTO.FTE;
-			newListing.CreatedDate = listingDTO.CreatedDate;
-			newListing.CreatedById = signedInUserId;
-
-
-			// Check if company exists
-			if (listingDTO.Employer is not null)
-			{
-				// TODO if exists 
-				var employer = await _companyRepository.GetAll().Where(e => e.Name == listingDTO.Employer).FirstOrDefaultAsync();
-				if (employer is null)
+				if (listingDTO.Employer is not null)
 				{
-					await _companyService.AddAsync(new CompanyDTO() { Name = listingDTO.Employer });
+					if (!await _companyService.ExistsByNameAsync(listingDTO.Employer))
+					{
+						await _companyService.AddAsync(new CompanyDTO() { Name = listingDTO.Employer });
+					}
+					newListingEntity.Employer = await _companyRepository.GetAll().Where(e => e.Name == listingDTO.Employer).FirstOrDefaultAsync();
 				}
 
-				newListing.Employer = await _companyRepository.GetAll().Where(e => e.Name == listingDTO.Employer).FirstOrDefaultAsync();
-			}
-
+				// Add sub entity Location
+				if (listingDTO.Location is not null)
+				{
+					if (!await _locationService.ExistsByNameAsync(listingDTO.Location))
+					{
+						await _locationService.AddAsync(new LocationDTO() { Name = listingDTO.Location });
+					}
+					newListingEntity.Location = await _locationRepository.GetAll().Where(e => e.Name == listingDTO.Location).FirstOrDefaultAsync();
+				}
 
-			// Check if location exists
-			if (listingDTO.Location is not null)
-			{
-				var location = await _locationRepository.GetAll().Where(e => e.Name == listingDTO.Location).FirstOrDefaultAsync();
-				if (location is null)
+				// Add sub entity ExperienceLevel
+				if (listingDTO.ExperienceLevel is not null)
 				{
-					await _locationService.AddAsync(new LocationDTO() { Name = listingDTO.Location });
+					if (!await _experienceLevelService.ExistsByNameAsync(listingDTO.ExperienceLevel))
+					{
+						await _experienceLevelService.AddAsync(new ExperienceLevelDTO() { Name = listingDTO.ExperienceLevel });
+					}
+					newListingEntity.ExperienceLevel = await _experienceLevelRepository.GetAll().Where(e => e.Name == listingDTO.ExperienceLevel).FirstOrDefaultAsync();
 				}
-				newListing.Location = await _locationRepository.GetAll().Where(e => e.Name == listingDTO.Location).FirstOrDefaultAsync();
 			}
-
-
-			// Check if experienceLevel exists
-			var experienceLevel = await _experienceLevelRepository.GetAll().Where(e => e.Name == listingDTO.ExperienceLevel).FirstOrDefaultAsync();
-			if (experienceLevel is null)
+			catch (Exception e)
 			{
-				// TODO - ExperienceLevel does not exist and need to be created.
+				_logger.LogError(e.Message);
 			}
-			newListing.ExperienceLevel = experienceLevel;
 
-
-			return newListing;
+			return newListingEntity;
 		}
 
-		// TODO - Should this return anything if string id is not found? Return a delete success/fail bool?
 		public async Task DeleteAsync(string id)
```

Comment:
Do we need to manage logic for cascade here in relation to Inquiries etc.?

Reply:
How about we add that as a separate issue so that we may test the current PR to see its current behaviour?

Reply:
Issue #78 created.